src/test/librbd/io/test_mock_ImageRequest.cc

```diff
@@ -367,6 +395,149 @@ TEST_F(TestMockIoImageRequest, DiscardGranularity) {
   ASSERT_EQ(0, aio_comp_ctx.wait());
 }
 
+TEST_F(TestMockIoImageRequest, EmptyDiscardJournalAppendEnabled) {
+  REQUIRE_FEATURE(RBD_FEATURE_JOURNALING);
+
+  librbd::ImageCtx *ictx;
+  ASSERT_EQ(0, open_image(m_image_name, &ictx));
+  ASSERT_EQ(0, resize(ictx, ictx->layout.object_size));
+  ictx->discard_granularity_bytes = 32;
+
+  MockTestImageCtx mock_image_ctx(*ictx);
+  librbd::MockTestPolicy policy;
+  mock_image_ctx.journal_policy = &policy;
+  MockTestJournal mock_journal;
+  mock_image_ctx.journal = &mock_journal;
+
+  InSequence seq;
+  expect_get_modify_timestamp(mock_image_ctx, false);
+  expect_is_journal_appending(mock_journal, true);
+  expect_journal_append_io_event_times(mock_journal, 0);
+  expect_object_request_send_times(mock_image_ctx, 0);
+
+  C_SaferCond aio_comp_ctx;
+  AioCompletion *aio_comp = AioCompletion::create_and_start(
+    &aio_comp_ctx, ictx, AIO_TYPE_DISCARD);
+  MockImageDiscardRequest mock_aio_image_discard(
+    mock_image_ctx, aio_comp,
+    {{16, 15}, {96, 31}},
```

Comment:
I wanted to test if (!object_extents.empty()) basically.

Reply:
I see. Let's drop `{16, 15}` and leave just `{96, 31}` then because currently both extents belong to the same "too short" category.  Also, I'd place this test before `DiscardGranularityJournalAppendEnabled` instead of here so that `discard_granularity_bytes = 32` tests are together.

Reply:
Changed to only {96, 31}. This of course works with the old code, since nothing is sent if object_extents is empty.  That may be a reason to drop it.